app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,41 @@
 class PostsController < ApplicationController
   def index
-    @users = User.find(params[:user_id])
-    @posts = @users.posts
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
   end
 
   def show
-    @users = User.find(params[:user_id])
-    @post = @users.posts.find(params[:id])
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+  end
+
+  def new
+    @user = current_user
+    @post = @user.posts.build
+  end
+
+  def create
+    @user = current_user
+    @post = Post.new(
+      author: @user,
+      title: params[:post][:title],
+      text: params[:post][:text],
+      comments_counter: 0,
+      likes_counter: 0
```

Comment:
- [ ] In trying to fix the error from the previous reviewer, you introduced another error. I cannot create a post due to misspelled attributes.

  These should be: `comment_counter` and `like_counter`, without the `s`.

  ```
  comment_counter: 0,
  like_counter: 0
  ```

  ![image](https://github.com/supreeti/BlogApp-ROR/assets/54848363/9159d9f5-7ea6-46e3-bc77-80557348d8a7)